presto-main/src/main/java/io/prestosql/version/EmbedVersion.java

```diff
@@ -102,4 +141,19 @@ public Runnable embedVersion(Runnable runnable)
             throw new RuntimeException(throwable);
         }
     }
+
+    @Override
+    public <T> Callable<T> embedVersion(Callable<T> callable)
+    {
+        requireNonNull(callable, "callable is null");
+        try {
+            @SuppressWarnings("unchecked")
+            Callable<T> wrapped = (Callable<T>) callableConstructor.invoke(callable);
+            return wrapped;
+        }
+        catch (Throwable throwable) {
+            throwIfUnchecked(throwable);
+            throw new RuntimeException(throwable);
```

Comment:
Please re-throw original exception

Reply:
`throwIfUnchecked(throwable)` does this (if possible)

note: this is only for creation of the callable; actual callable will propagate exceptions just fine